### ISSUE TITLE
Screen reader is not announcing the selection state of the "Violations"/"Passes"/"Incomplete" tab items in Accessibility Tab.

### ISSUE THREAD

**Issue by msftedad** [storybookjs/storybook#26674]
### Describe the bug

OS: Windows 11
OS version: 23H2 (OS Build 22631.3296)
Edge Version: 122.0.2365.80 (Official build) (64-bit)

### To Reproduce

1.Open the URL " [Overview - Docs ⋅ Storybook (windows.net)](https://nam06.safelinks.protection.outlook.com/?url=https:%2f%2fvivacommonstorybooktest.blob.core.windows.net%2f%24web%2findex.html?path%3D%2fdocs%2foverview--docs&data=05%7c02%7cv-subaku%40microsoft.com%7c1d03a870859b492ba99f08dbf6498987%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c638374567927399114%7cUnknown%7cTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7c3000%7c%7c%7c&sdata=924Ca%2B3zJMJHXpDXaCwWOTu%2fHX1W%2fJ2TKRp6YoSBli8%3D&reserved=0)".
2.Navigate to the "Tab list" in the left pane and expand it.
3.Navigate to "default" and invoke it.
4.Navigate to "Accessibility" tab item and invoke it.
5.Navigate to the " violations, passes, incomplete tab items.
6.Observe the narrator announcement.

[b12dd9a1-4c07-4f73-8204-4a056297188d.webm](https://github.com/storybookjs/storybook/assets/93735775/32c849c4-9520-4964-89ac-16ac90928e5f)

### System

```bash
Actual Results: 

While navigating to the " violations, passes, incomplete tab items, narrator is not announcing the selected/unselected state.

Expected Results: 

While navigating to the " violations, passes, incomplete” tab items, narrator should announce the selected/unselected state.
```

**Comment by Mohith1490** [comment]
@msftedad I'm mohith, I'm a new contributor can you give the file path for the issue

**Comment by JohannesFischer** [comment]
I just worked on a Storybook addon with a very similar layout and would be happy to help out with this issue.

**Comment by WioletaKolodziej** [comment]
Has this issue been fixed yet, is it still reproducible?

**Comment by Sidnioulz** [comment]
A component that fixes this is planned in https://github.com/storybookjs/storybook/issues/32235. It can then be applied to the a11y addon.
